src/java/org/apache/cassandra/security/PEMBasedSslContextFactory.java

```diff
@@ -120,6 +123,23 @@ else if (!StringUtils.isEmpty(keystore_password) && !keyPassword.equals(keystore
                         "okay. Ideally you should only specify one of them.");
         }
 
+        pemEncodedOutboundKey = StringUtils.defaultString(getString(ConfigKey.OUTBOUND_KEY.getKeyName()), pemEncodedKey);
+        outboundKeyPassword = StringUtils.defaultString(getString(ConfigKey.OUTBOUND_KEY_PASSWORD.getKeyName()), keyPassword);
+        if (StringUtils.isEmpty(outboundKeyPassword))
+        {
+            outboundKeyPassword = outbound_keystore_password;
+        }
+        else if (!StringUtils.isEmpty(outboundKeystore) && !outboundKeyPassword.equals(outbound_keystore_password))
+        {
+            throw new IllegalArgumentException("'outbound_keystore_password' and 'outbound_key_password' both configurations are given and the " +
+                                               "values do not match");
+        }
+        else
+        {
+            logger.warn("'keystore_password' and 'key_password' both are configured but since the values match it's " +
```

Comment:
should the warning about `outbound_keystore_password` and `outbound_key_password` instead? The `if-elseif-else` logic basically duplicates the above. It could use a refactor.